src/lexer.c

```diff
@@ -66,47 +66,84 @@ static bool IsString() {
 
   token.type = kTokenString;
 
+  token.precedence = kPrecPrimary;
+
   return true;
 }
 
 static bool IsCharacter() {
   switch (*source_code) {
     case '=':
-      token.type = kTokenEquals;
+      source_code++;
+      if (*source_code == '=') {
+        token.type = kTokenEquals;
+        token.precedence = kPrecComparison;
+        // NOLINTNEXTLINE(clang-analyzer-security.insecureAPI.DeprecatedOrUnsafeBufferHandling)
+        strncpy(token.value.text, "==", 3);
```

Comment:
@WannabeMO We stopped copying everything except string literals or identifiers to `token.value.text` to save a few clock cycles. Setting `token.type` should be enough to distinguish the token from other tokens. The same cannot be said about strings or identifiers, where the literal value distinguishes them from other strings or identifiers. so you can remove lines 81 and 82.

```suggestion
```